restapi/api/device_endpoints.go

```diff
@@ -293,3 +302,57 @@ func DisableDeviceCondition(c *gin.Context) {
 
 	c.JSON(http.StatusOK, GenericResponse{Message: "Device condition disabled"})
 }
+
+//========================================
+// DEVICE PAIRING
+//========================================
+// Pairs a device
+// @Summary      Pair a device with/without supervision
+// @Description  Pair a device with/without supervision
+// @Tags         general_device_specific
+// @Produce      json
+// @Success      200  {object}  GenericResponse
+// @Failure      500  {object}  GenericResponse
+// @Failure      422  {object}  GenericResponse
+// @Param        udid path string true "Device UDID"
+// @Param        supervised query string true "Set if device is supervised - true/false"
+// @Param 		 p12file formData file false "Supervision *.p12 file"
+// @Param 		 supervision_password formData string false "Supervision password"
+// @Router       /device/{udid}/pair [post]
+func PairDevice(c *gin.Context) {
+	device := c.MustGet(IOS_KEY).(ios.DeviceEntry)
+
+	supervised := c.Query("supervised")
+	if supervised == "" {
+		c.JSON(http.StatusUnprocessableEntity, GenericResponse{Error: "supervised query param is missing (true/false)"})
+		return
+	}
+
+	if supervised == "false" {
+		err := ios.Pair(device)
+		if err != nil {
+			c.JSON(http.StatusInternalServerError, GenericResponse{Error: err.Error()})
+			return
+		}
+		c.JSON(http.StatusOK, GenericResponse{Message: "Device paired"})
+		return
+	}
+
+	file, _, err := c.Request.FormFile("p12file")
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, GenericResponse{Error: "Could not parse p12 file from form-data, err:" + err.Error()})
+		return
+	}
+	p12fileBuf := new(bytes.Buffer)
+	p12fileBuf.ReadFrom(file)
+
+	supervision_password := c.Request.FormValue("supervision_password")
```

Comment:
could also go in as a header param. maybe nicer than a form parameter

Reply:
I was wondering which path to take, this is not production ready so security is not an issue xD having it in a header is as good as any, I'll change it

Reply:
It's less security, more UX. I think putting it in the header might be easier to do because it already has key value format. and than you can just post the file which should simplify caller code.

Reply:
moved password to a header, instead of part of form-data